File: day-8/solution.py
class Solution:
	def Game(self, stacks: List[int]) -> bool:
    	a = l = 0
    	while(stacks):
        	if stacks[0]>=stacks[-1]:
            	a += stacks.pop(0)
            	l += stacks.pop(-1)
        	else:
            	a += stacks.pop(-1)
            	l += stacks.pop(0)
    	return a>l

File: day-8/solution.cpp
class Solution {
public:
	bool Game(vector<int>& stacks) {
    	int Mansi_count=0,riya_count=0,n=stacks.size(),last=n-1,first=0;
    	bool chance=true;
   	 
    	while(n!=0)
    	{
        	if(chance)
        	{
            	Mansi_count+=max(stacks[last],stacks[first]);
            	if(stacks[first]>=stacks[last]){first++;}
            	else{last--;}
        	}
        	else
        	{
            	riya_count+=min(stacks[last],stacks[first]);
            	if(stacks[first]>=stacks[last]){last--;}
            	else{first++;}
        	}
        	n--;
        	chance=(!chance);
    	}
    	return Mansi_count>riya_count;
	}
};
